File: airportgap-typescript/tests/favorites/favoritesCrud.test.ts
import { describe, it, expect } from 'vitest'
import { randomUUID } from 'node:crypto'
import { Env } from '../../src/config/env.js'
import { ApiClient } from '../../src/http/apiClient.js'

describe('Favorites CRUD', () => {
  const api = new ApiClient()

  it('create, read, update, delete', async function () {
    if (!Env.token) {
      console.warn('Skipping favorites test: TOKEN not set')
      return
    }

    // create
    const note = `created-by-tests-${randomUUID()}`
    const create = await api.auth.post('/favorites', { airport_id: 'JFK', note })
    expect(create.status).toBe(201)
    const id = create.data?.data?.id
    expect(id).toBeTruthy()

    // read list
    const list = await api.auth.get('/favorites')
    expect(list.status).toBe(200)
    const ids = (list.data?.data ?? []).map((x: any) => x.id)
    expect(ids).toContain(id)

    // update
    const newNote = `${note}-updated`
    const upd = await api.auth.patch(`/favorites/${id}`, { note: newNote })
    expect(upd.status).toBe(200)
    const one = await api.auth.get(`/favorites/${id}`)
    expect(one.data?.data?.attributes?.note).toBe(newNote)

    // delete
    const del = await api.auth.delete(`/favorites/${id}`)
    expect(del.status).toBe(204)
  })
})

File: airportgap-java/src/test/java/spec/favorites/FavoritesCrudTest.java
package spec.favorites;

import config.Env;
import http.ApiClient;
import org.junit.jupiter.api.Assumptions;
import org.junit.jupiter.api.Test;
import java.util.UUID;
import static io.restassured.RestAssured.given;
import static org.assertj.core.api.Assertions.assertThat;

public class FavoritesCrudTest {
  private final ApiClient api = new ApiClient(Env.baseUrl(), Env.token());

  @Test
  void createReadUpdateDeleteFavorite() {
    Assumptions.assumeTrue(Env.token() != null && !Env.token().isBlank(), "TOKEN must be set for favorites tests");

    // create
    String note = "created-by-tests-" + UUID.randomUUID();
    var createResp = given()
        .spec(api.authSpec())
        .header("Content-Type", "application/json")
        .body("{\"airport_id\":\"JFK\", \"note\":\""+note+"\"}")
        .post("/favorites")
        .then()
        .statusCode(201)
        .extract().jsonPath();

    String favId = createResp.getString("data.id");
    assertThat(favId).isNotBlank();

    // read
    var list = given().spec(api.authSpec())
        .get("/favorites")
        .then().statusCode(200)
        .extract().jsonPath().getList("data");
    assertThat(list.toString()).contains(favId);

    // update
    String newNote = note + "-updated";
    given().spec(api.authSpec())
        .header("Content-Type", "application/json")
        .body("{\"note\":\""+newNote+"\"}")
        .patch("/favorites/" + favId)
        .then().statusCode(200);

    var getOne = given().spec(api.authSpec())
        .get("/favorites/" + favId)
        .then().statusCode(200)
        .extract().jsonPath();
    assertThat(getOne.getString("data.attributes.note")).isEqualTo(newNote);

    // delete
    given().spec(api.authSpec())
        .delete("/favorites/" + favId)
        .then().statusCode(204);
  }
}
